lms/static/scripts/frontend_apps/components/StudentSelector.tsx

```diff
@@ -0,0 +1,120 @@
+import {
+  ArrowLeftIcon,
+  ArrowRightIcon,
+  IconButton,
+  InputGroup,
+  Select,
+} from '@hypothesis/frontend-shared/lib/next';
+import classnames from 'classnames';
+
+type Student = {
+  displayName: string;
+};
+
+export type StudentSelectorProps = {
+  /** Callback invoked when the selected student changes */
+  onSelectStudent: (index: number) => void;
+  /** Index of selected student in `students` or -1 if no student is selected */
+  selectedStudentIndex: number;
+  /** Ordered list of students to display in the drop-down */
+  students: Student[];
+};
+
+/**
+ * A student navigation tool which shows a student selection list and a previous and next button.
+ */
+export default function StudentSelector({
+  onSelectStudent,
+  selectedStudentIndex,
+  students,
+}: StudentSelectorProps) {
+  // Disable the next button if at the end of the list. The length is equal to
+  // the student list plus the default "All Students" option.
+  const hasNextView = selectedStudentIndex + 1 < students.length;
+  // Disable the previous button only if the selectedStudentIndex is less than 0
+  // indicating the "All Students" choice is selected.
+  const hasPrevView = selectedStudentIndex >= 0;
+
+  /**
+   * Select the next student index in the list.
+   */
+  const onNextView = () => {
+    onSelectStudent(selectedStudentIndex + 1);
+  };
+  /**
+   * Select the previous student index in the list. The 0 index
+   * represents "All Students."
+   */
+  const onPrevView = () => {
+    onSelectStudent(selectedStudentIndex - 1);
+  };
+
+  return (
+    <div
+      className={classnames(
+        // Narrower widths: label above field
+        'flex flex-col gap-1',
+        // Wider widths: label to left of field
+        'xl:flex-row xl:gap-3 xl:items-center'
+      )}
+    >
+      <label
+        className={classnames(
+          'flex-grow font-medium text-sm leading-none',
+          'xl:text-right'
+        )}
+        data-testid="student-selector-label"
+      >
+        {selectedStudentIndex >= 0 ? (
+          <span>
+            Student {`${selectedStudentIndex + 1} of ${students.length}`}
+          </span>
+        ) : (
+          <span>{`${students.length} Students`}</span>
+        )}
+      </label>
+      <div>
+        <InputGroup>
+          <IconButton
+            icon={ArrowLeftIcon}
+            title="previous student"
```

Comment:
```suggestion
            title="Previous student"
```

These titles are user visible in the form of tooltips.